src/platforms/hosted/windows/ftdi.c

```diff
@@ -149,11 +149,22 @@ int ftdi_read_data(struct ftdi_context *ftdi, unsigned char *buf, const int size
 
 int ftdi_write_data(struct ftdi_context *ftdi, const unsigned char *buf, int size)
 {
-	(void)ftdi;
-	DWORD bytes_written;
-	if (FT_Write(ftdi_handle, (unsigned char *)buf, size, &bytes_written) != FT_OK)
```

Comment:
This line should only need wrapping with
```c
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wcast-qual"
...
#pragma GCC diagnostic pop
```
Not sure the rest of this change should be necessary and it massively pessimises performance if it is, so we want to avoid it if possible.